pkg/codegen/nodejs/gen.go

```diff
@@ -790,11 +790,10 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) (resourceFil
 
 					arg = fmt.Sprintf("(%s) ?? %s", arg, dv)
 				}
+			}
 
-				// provider properties must be marshaled as JSON strings.
-				if r.IsProvider && !isStringType(prop.Type) {
-					arg = fmt.Sprintf("pulumi.output(%s).apply(JSON.stringify)", arg)
-				}
+			if r.IsProvider && !isStringType(prop.Type) {
+				arg = fmt.Sprintf("pulumi.output(%s)", arg)
 			}
```

Comment:
I don't think we need to keep this. We were only wrapping in `pulumi.output` so we could call `apply`.

Reply:
Yeah, I was intentionally trying to scope this to the absolute bare minimum changes but I will try pulling this out.